src/fontra/client/core/utils.js

```diff
@@ -353,6 +353,44 @@ export function rgbaToCSS(rgba) {
   return `rgb(${channels.join(",")})`;
 }
 
+export function hexToRgba(hexColor) {
+  let c = hexColor.substring(1).split("");
+  let r = [];
+  if (/^#[A-Fa-f0-9]{8}$/.test(hexColor) || /^#[A-Fa-f0-9]{6}$/.test(hexColor)) {
+    for (const i of range(0, c.length, 2)) {
+      r.push(round(parseInt(c[i] + c[i + 1], 16) / 255, 2));
+    }
+  } else if (/^#[A-Fa-f0-9]{4}$/.test(hexColor) || /^#[A-Fa-f0-9]{3}$/.test(hexColor)) {
+    for (const i of range(c.length)) {
+      r.push(round(parseInt(c[i] + c[i], 16) / 255, 2));
+    }
+  } else {
+    r = [1, 0, 0, 1];
+    new Error("Bad hex color format. Should be #RRGGBB or #RRGGBBAA or #RGB or #RGBA");
+  }
+  if (r.length === 3) {
+    r.push(1);
+  }
+  return r;
+}
+
+export function rgbaToHex(rgba) {
+  const channels = rgba.slice(0, 3).map((channel) =>
```

Comment:
Suggest to throw an error if rgba.length is not 3 or 4.

Then, I would not do the slice, and later check the fourth element, and remove it if it is "ff". This would remove the need to repeat the hex conversion / zero padding code.

Reply:
I have a solution, but I am not 100% sure if it's correct. Why I keep this open for now.